compiler-rt/test/nsan/nan.cpp

```diff
@@ -0,0 +1,19 @@
+// RUN: %clangxx_nsan -O0 -g %s -o %t
+// RUN: NSAN_OPTIONS=check_nan=true %run %t 2>&1 | FileCheck %s
```

Comment:
Could you please also test halt_on_error=0 / halt_on_error=1 here ? (similarly to the other test)